samples/highcharts/demo/calendar-heatmap-month/demo.js

```diff
@@ -0,0 +1,293 @@
+    temperature: 11.2
+},
+{
+    date: '2017-05-23',
+    temperature: 13.5
+},
+{
+    date: '2017-05-24',
+    temperature: 11.0
+},
+{
+    date: '2017-05-25',
+    temperature: 10.1
+},
+{
+    date: '2017-05-26',
+    temperature: 11.7
+},
+{
+    date: '2017-05-27',
+    temperature: 15.9
+},
+{
+    date: '2017-05-28',
+    temperature: 16.8
+},
+{
+    date: '2017-05-29',
+    temperature: 13.1
+},
+{
+    date: '2017-05-30',
+    temperature: 12.8
+},
+{
+    date: '2017-05-31',
+    temperature: 11.7
+}];
+
+const weekdays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+
+// The function takes in a dataset and calculates how many empty tiles needed before and after the dataset is plotted.
+function generateChartData(data) {
+
+    // Calculate the starting weekday index (0-6 of the first date in the given array)
+    const firstWeekday = new Date(data[0].date).getDay(),
+        monthLength = data.length,
+        lastElement = data[monthLength - 1].date,
+        lastWeekday = new Date(lastElement).getDay(),
+        lengthOfWeek = 6,
+        emptyTilesFirst = firstWeekday,
+        chartData = [];
+
+    // Add the empty tiles before the first day of the month with null values to take up space in the chart
+    for (let emptyDay = 0; emptyDay < emptyTilesFirst; emptyDay++) {
+        chartData.push({
+            x: emptyDay,
+            y: 4,
+            value: null,
+            date: null,
+            custom: {
+                empty: true
+            }
+        });
+    }
+
+    // Loop through and populate with temperature and dates from the dataset
+    for (let day = 1; day <= monthLength; day++) {
+        const date = data[day - 1].date; // Get the date from the given data array
+        const xCoordinate = (emptyTilesFirst + day - 1) % 7; // Offset by the number of empty tiles
+        const yCoordinate = Math.floor((firstWeekday + day - 1) / 7);
+        const id = day;
+
+        // Get the corresponding temperature for the current day from the given array
+        const temperature = data[day - 1].temperature;
+
+        chartData.push({
+            x: xCoordinate,
+            y: 4 - yCoordinate,
+            value: temperature,
+            date: new Date(date).getTime(),
+            custom: {
+                monthDay: id
+            }
+        });
+    }
+
+    // Fill in the missing values when dataset is looped through.
+    const emptyTilesLast = lengthOfWeek - lastWeekday;
+    for (let emptyDay = 1; emptyDay <= emptyTilesLast; emptyDay++) {
+        chartData.push({
+            x: (lastWeekday + emptyDay) % 7,
+            y: 0,
+            value: null,
+            date: null,
+            custom: {
+                empty: true
+            }
+        });
+    }
+    return chartData;
+}
+const chartData = generateChartData(data);
+
+Highcharts.chart('container', {
+    chart: {
+        type: 'heatmap'
+    },
+
+    title: {
+        text: 'Mid-day temperature May 2017',
```

Comment:
I think we should find a newer data set. People looking at this will subconciously think that the demo hasn't been updated since 2017. Actually we have been trying to systematically keep our demo data updated to avoid that.